api/v1alpha1/common.go

```diff
@@ -221,34 +221,45 @@ func filterMatchVersionConstraint(cond FilterCond, filter Filter) bool {
 	return false
 }
 
-func Match(filterCond map[string]FilterCond, filter Filter) bool {
-	if len(filterCond) == 0 {
-		return true
-	}
-
-	cond, ok := filterCond[filter.Name]
-	if !ok {
-		return true
+// Match determines if this component is retained, and if so, filters for conforming versions.
+func Match(fc map[string]FilterCond, filter Filter, funcs ...FilterFunc) ([]int, bool) {
+	var versions []int
+	if len(funcs) == 0 {
+		funcs = []FilterFunc{FilterMatchVersion, FilterMatchVersionRegexp, FilterMatchVersionConstraint}
 	}
-
-	if !filterMatchDeprecation(cond, filter) {
-		return false
-	}
-	if filterMatchVersion(cond, filter) {
-		return true
+	if len(fc) == 0 {
+		for i := range filter.Versions {
+			versions = append(versions, i)
+		}
+		return versions, true
 	}
-	if filterMatchRegexp(cond, filter) {
-		return true
+	filterCond, ok := fc[filter.Name]
+	if ok && filterCond.Operation == FilterOpIgnore {
```

Comment:
considering to make this as a common `FilterFunc` too

Reply:
This piece of code is used to check if the chart is retained, and the FilterFunc below is used to determine if the version is retained.